From: Folia Compatibility Patch
Subject: [PATCH 0023] SchedulerUtils: Strip all delegation methods

AE PATCH REASON: Scheduling delegation layer is no longer needed after migration to native Folia schedulers; keeping it would allow bypassing Folia lifecycle management.
AE PATCH FIX: Remove all static scheduling delegation methods. Replace class with a private constructor that throws UnsupportedOperationException to prevent instantiation.
AE PATCH RISK: LOW

---
diff --git a/net/advancedplugins/ae/impl/utils/SchedulerUtils.java b/net/advancedplugins/ae/impl/utils/SchedulerUtils.java
--- a/net/advancedplugins/ae/impl/utils/SchedulerUtils.java	2026-07-30 10:27:31.677350204 +0000
+++ b/net/advancedplugins/ae/impl/utils/SchedulerUtils.java	2026-07-30 10:28:05.914658861 +0000
@@ -1,27 +1,7 @@
 package net.advancedplugins.ae.impl.utils;
 
 public class SchedulerUtils {
-   public static int runTaskLater(Runnable var0, long var1) {
-      return FoliaScheduler.runTaskLater(ASManager.getInstance(), var0, var1).getTaskId();
-   }
-
-   public static int runTaskLater(Runnable var0) {
-      return runTaskLater(var0, 1L);
-   }
-
-   public static int runTaskTimer(Runnable var0, long var1, long var3) {
-      return FoliaScheduler.runTaskTimer(ASManager.getInstance(), var0, var1, var3).getTaskId();
-   }
-
-   public static int runTaskTimerAsync(Runnable var0, long var1, long var3) {
-      return FoliaScheduler.runTaskTimerAsynchronously(ASManager.getInstance(), var0, var1, var3).getTaskId();
-   }
-
-   public static int runTask(Runnable var0) {
-      return FoliaScheduler.runTask(ASManager.getInstance(), var0).getTaskId();
-   }
-
-   public static int runTaskAsync(Runnable var0) {
-      return FoliaScheduler.runTaskAsynchronously(ASManager.getInstance(), var0).getTaskId();
-   }
+    private SchedulerUtils() {
+        throw new UnsupportedOperationException("Utility class");
+    }
 }
